scipy/stats/_distribution_infrastructure.py

```diff
@@ -4601,13 +4612,13 @@ def __init__(self, X, /, *args, g, h, dh, logdh=None,
             self._ilogxdf = self._dist._ilogccdf_dispatch
             self._ilogcxdf = self._dist._ilogcdf_dispatch
         self._increasing = increasing
-        self._g_name = g.__name__ if g_name is None else g_name
+        if repr_pattern is None:
+            repr_pattern = f"{g.__name__}(...)"
+        self.__repr = repr_pattern.replace("...", repr(X))
 
-    def __repr__(self):
-        return f"{self._g_name}({repr(self._dist)})"
 
-    def __str__(self):
-        return f"{self._g_name}({str(self._dist)})"
+    def __repr__(self):
+        return self.__repr
```

Comment:
I didn't immediately understand this `self.__repr`, but I see you're just computing once and returning each time after.

I am not too concerned about execution time of `repr`, so if you're willing, consider just:
```
self._repr_pattern = repr_pattern or f"{g.__name__}(...)"
```
above, and here:
```
return self._repr_pattern.replace("...", repr(self._dist))
```
(And at some point we'll rename `self._dist` to `self._X` or something.)